rewrite-java/src/main/java/org/openrewrite/java/ReplaceUUIDRandomId.java

```diff
@@ -0,0 +1,89 @@
+package org.openrewrite.java;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.jetbrains.annotations.NotNull;
+import org.openrewrite.*;
+import org.openrewrite.config.RecipeDescriptor;
+import org.openrewrite.java.tree.J;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import org.openrewrite.internal.lang.NonNull;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class ReplaceUUIDRandomId extends Recipe {
+    @Option(displayName = "Fully Qualified Class Name",
+            description = "A fully qualified class name indicating in which class to replace UUId.randomUUID() method to Tree.randomId() .",
+            example = "com.yourorg.FooBar")
+    @NonNull
+    String fullyQualifiedClassName;
+
+    @Override
+    public String getDisplayName() {
+        return "Replace UUID.randomUUID() with Tree.randomId()";
+    }
+
+    @Override
+    public String getDescription() {
+        return "This recipe replaces occurrences of UUID.randomUUID() with Tree.randomId().";
+    }
+
+    @JsonCreator
+    public ReplaceUUIDRandomId(@NonNull @JsonProperty("fullyQualifiedClassName") String fullyQualifiedClassName) {
+        this.fullyQualifiedClassName = fullyQualifiedClassName;
+    }
+
+    @Override
+    public Set<String> getTags() {
+        return Collections.singleton("uuid-replacement");
+    }
+
+    @Override
+    public Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(2);
+    }
+
+    @Override
+    public RecipeDescriptor createRecipeDescriptor() {
+        // Customize if needed
+        return super.createRecipeDescriptor();
+    }
+
+    @Override
+    public @NotNull List<Recipe> getRecipeList() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public @NotNull TreeVisitor<J, ExecutionContext> getVisitor() {
+        return new ReplaceUUIDRandomIdVisitor();
+    }
+
+    private static class ReplaceUUIDRandomIdVisitor extends JavaIsoVisitor<ExecutionContext> {
+        @Override
+        public J.Literal visitLiteral(J.Literal literal, ExecutionContext ctx) {
+            J.Literal l = super.visitLiteral(literal, ctx);
```

Comment:
Instead of `visitLiteral` you'll want to override `visitMethodInvocation`, then use a MethodMatcher to check that it's a LST element constructor call, and if is then inspect the first argument, and if that is a call the UUID.randomUUID, then replace that first argument using `ListUtils.mapFirst`. Does that help?